### PR TITLE
Fixing build info in binaries

### PR DESCRIPTION
**Pull request by evanxg852000**
https://github.com/quickwit-oss/quickwit/issues/2531

### CODE REVIEW

File: .github/actions/build-binary/action.yml [hunk 1]
```diff
@@ -30,6 +30,9 @@ runs:
     - name: Install cross
       run: cargo install cross
       shell: bash
+    - name: Help cross find the git repo
+      run: cp -r .git/ ./quickwit/.git
```
Comment (guilload) [inline .github/actions/build-binary/action.yml:34]: Use a symlink.

Comment (evanxg852000) [reply .github/actions/build-binary/action.yml:34]: Yes much better. I just wanted something radical given the time it takes to test it. Also, this ends up mounted in a docker container.

Comment (guilload) [reply .github/actions/build-binary/action.yml:34]: Can't we use passthrough env variables for `cross`?
https://github.com/cross-rs/cross#passing-environment-variables-into-the-build-environment

Comment (evanxg852000) [reply .github/actions/build-binary/action.yml:34]: Yes, it was used at some point

Comment (guilload) [reply .github/actions/build-binary/action.yml:34]: The build script is nop if git is not available in the env. `QuickwitBuildInfo` attempts to read the `QW_COMMIT_*` en vars directly. So as long as they are populated by the build script or whatever process, we're good.

**Pull request closed without merging**